Src/Plugins/Security/SEC101_050.IdentifiableNpmAuthorTokenValidator.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk;
+using Microsoft.RE2.Managed;
+
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class IdentifiableNpmAuthorTokenValidator : DynamicValidatorBase
+    {
+        [ThreadStatic]
+        private static StringBuilder s_sb;
+
+        protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary<string, FlexMatch> groups)
+        {
+            FlexMatch secret = groups["secret"];
+
+            if (groups.TryGetNonEmptyValue("checksum", out FlexMatch checksum))
```

Comment:
>if (groups.TryGetNonEmptyValue("checksum", out FlexMatch checksum)) [](http://example.com/codeflow?start=12&length=67)

since this is required, you can follow what we did for the secret. If we don't have that key, it will throw an exception because that is not expected.